kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/IWAPParser.java

```diff
@@ -34,7 +35,12 @@
     private IWSecuritySectionParser securityParser = null;
 
     public IWAPParser(String macAddressString) {
-        this.hardwareAddress = NetworkUtil.macToBytes(macAddressString);
+        try {
+            this.hardwareAddress = NetworkUtil.macToBytes(macAddressString);
+        } catch (KuraException e) {
+            // Hardware address is invalid
+            s_logger.info("Invalid hardware address");
```

Comment:
With switching to use of IllegalArgumentException, changes in other classes are no longer needed. All changes in classes other than NetworkUtil will be reverted.